Classes/Controller/TableOfContentsController.php

```diff
@@ -176,7 +176,7 @@ public function makeMenuArray()
         // Does the document have physical elements or is it an external file?
         if (
             !empty($this->document->getDoc()->physicalStructure)
-            || !MathUtility::canBeInterpretedAsInteger($this->document->getDoc()->uid)
+            || !MathUtility::canBeInterpretedAsInteger($this->requestData['id'])
```

Comment:
Interesting question.

I would assume that XML file content has in that case higher priority than this what is stored in the database.

Is it also not specified in some other place (creating document instance?) that if URL points into the database object it transforms from URL to numerical ID?

Reply:
That's a good question and we should find a solution for the TOC anyway. But this is buggy in 3.x (and even 2.x) I think.

Example:
1. call Börsenblatt anchor: http://digital.slub-dresden.de/id39946221X
2. in TOC: all years are listed with IDs --> expected behaviour
3. click "1834" --> all issues have IDs but the backlink to the anchor has an URL now
4. click the anchor link --> you are back in 1.) with the difference, that
   * the link to itself is via URL
   * you cannot search in the anchor (full title) because no ID is given 
   
Things get worse, if you have only e.g. a part of the years indexed locally. In this case in step 2.) all years are listed because they are part of the METS file and they are linked via URL.

We cannot fix this in this PR and should think about in a new issue.